proxy/tests/transparency.rs

```diff
@@ -311,5 +319,68 @@ fn http1_get_doesnt_add_transfer_encoding() {
         .inbound(srv)
         .run();
     let client = client::http1(proxy.inbound, "transparency.test.svc.cluster.local");
-    assert_eq!(client.get("/"), "hello h1");
+
+    let methods = &[
+        "GET",
+        "POST",
+        "PUT",
+        "DELETE",
+        "HEAD",
+        "PATCH",
+    ];
+
+    for &method in methods {
+        let resp = client.request(
+            client
+                .request_builder("/")
+                .method(method)
+        );
+
+        assert_eq!(resp.status(), StatusCode::OK, "method={:?}", method);
+    }
+}
+
+#[test]
+fn http1_content_length_zero_is_preserved() {
+    let _ = env_logger::try_init();
+
+    let srv = server::http1()
+        .route_fn("/", |req| {
+            let status = if req.headers()["content-length"] == "0" {
+                StatusCode::OK
+            } else {
+                StatusCode::BAD_REQUEST
+            };
+            let mut res = Response::new("".into());
+            *res.status_mut() = status;
+            res
+        })
+        .run();
+    let ctrl = controller::new().run();
+    let proxy = proxy::new()
+        .controller(ctrl)
+        .inbound(srv)
+        .run();
+    let client = client::http1(proxy.inbound, "transparency.test.svc.cluster.local");
+
+
+    let methods = &[
+        "GET",
+        "POST",
+        "PUT",
+        "DELETE",
+        "HEAD",
+        "PATCH",
+    ];
+
+    for &method in methods {
+        let resp = client.request(
+            client
+                .request_builder("/")
+                .method(method)
+                .header("content-length", "0")
+        );
+
+        assert_eq!(resp.status(), StatusCode::OK, "method={:?}", method);
+    }
```

Comment:
OK, great. Can we also do this same test, but using an empty body with `Transfer-Encoding: chunked`?

Finally, could you file a follow-up issue about making the same tests, except for response bodies. That is, we should test that we preserve `Content-Length: 0` in response bodies and we shouldn't convert a non-chunked response body to a chunked one (unintentionally). (Or, add the tests to this PR, if you want.)